site/src/pages/UserSettingsPage/SchedulePage/ScheduleForm.tsx

```diff
@@ -0,0 +1,609 @@
+  "Europe/Saratov",
+  "Europe/Simferopol",
+  "Europe/Sofia",
+  "Europe/Stockholm",
+  "Europe/Tallinn",
+  "Europe/Tirane",
+  "Europe/Ulyanovsk",
+  "Europe/Uzhgorod",
+  "Europe/Vienna",
+  "Europe/Vilnius",
+  "Europe/Volgograd",
+  "Europe/Warsaw",
+  "Europe/Zaporozhye",
+  "Europe/Zurich",
+  "Indian/Chagos",
+  "Indian/Christmas",
+  "Indian/Cocos",
+  "Indian/Kerguelen",
+  "Indian/Mahe",
+  "Indian/Maldives",
+  "Indian/Mauritius",
+  "Indian/Reunion",
+  "Pacific/Apia",
+  "Pacific/Auckland",
+  "Pacific/Bougainville",
+  "Pacific/Chatham",
+  "Pacific/Chuuk",
+  "Pacific/Easter",
+  "Pacific/Efate",
+  "Pacific/Enderbury",
+  "Pacific/Fakaofo",
+  "Pacific/Fiji",
+  "Pacific/Funafuti",
+  "Pacific/Galapagos",
+  "Pacific/Gambier",
+  "Pacific/Guadalcanal",
+  "Pacific/Guam",
+  "Pacific/Honolulu",
+  "Pacific/Kiritimati",
+  "Pacific/Kosrae",
+  "Pacific/Kwajalein",
+  "Pacific/Majuro",
+  "Pacific/Marquesas",
+  "Pacific/Nauru",
+  "Pacific/Niue",
+  "Pacific/Norfolk",
+  "Pacific/Noumea",
+  "Pacific/Pago_Pago",
+  "Pacific/Palau",
+  "Pacific/Pitcairn",
+  "Pacific/Pohnpei",
+  "Pacific/Port_Moresby",
+  "Pacific/Rarotonga",
+  "Pacific/Tahiti",
+  "Pacific/Tarawa",
+  "Pacific/Tongatapu",
+  "Pacific/Wake",
+  "Pacific/Wallis",
+]
+
+const getAllTimezones = () => {
+  if (Intl.supportedValuesOf) {
+    return Intl.supportedValuesOf("timeZone").sort()
+  }
+
+  return allTimezonesFallback
+}
+
+const getPreferredTimezone = () => {
+  return Intl.DateTimeFormat().resolvedOptions().timeZone
+}
+
+const timeToCron = (hours: number, minutes: number, tz: string) => {
+  let prefix = ""
+  if (tz !== "") {
+    prefix = `CRON_TZ=${tz} `
+  }
+  return `${prefix}${minutes} ${hours} * * *`
+}
+
+// evaluateNextRun returns a Date object of the next cron run time.
+const evaluateNextRun = (
+  hours: number,
+  minutes: number,
+  tz: string,
+  now: Date | undefined,
+): Date => {
+  // The cron-parser package doesn't accept a timezone in the cron string, but
+  // accepts it as an option.
+  const cron = timeToCron(hours, minutes, "")
+  const parsed = cronParser.parseExpression(cron, {
+    currentDate: now,
+    iterator: false,
+    utc: false,
+    tz: tz,
+  })
+
+  return parsed.next().toDate()
+}
+
+const formatNextRun = (
+  hours: number,
+  minutes: number,
+  tz: string,
+  now: Date | undefined,
+): string => {
+  const nowDjs = dayjs(now).tz(tz)
+  const djs = dayjs(evaluateNextRun(hours, minutes, tz, now)).tz(tz)
+  let str = djs.format("h:mm A")
+  if (djs.isSame(nowDjs, "day")) {
+    str += " today"
+  } else if (djs.isSame(nowDjs.add(1, "day"), "day")) {
+    str += " tomorrow"
+  } else {
+    // This case will rarely ever be hit, as we're dealing with only times and
+    // not dates, but it can be hit due to mismatched browser timezone to cron
+    // timezone or due to daylight savings changes.
+    str += ` on ${djs.format("dddd, MMMM D")}`
+  }
+
+  str += ` (${djs.from(now)})`
+
+  return str
+}
```

Comment:
this feels like the kind of thing that dayjs should just support. does it not?

Reply:
I don't think you can do complex formatting like this. This is `h:mm A [today|tomorrow|on dddd, MMMM D] (in 12 hours)`